File: test/leetcode/tree/diameter-of-binary-tree.test.ts
import { array2tree } from '../../src/tree/common/tree-node';
import { diameterOfBinaryTree } from '../../src/tree/diameter-of-binary-tree';

describe('diameter of binary tree', () => {
  test('diameter of binary tree - test case 1', async () => {
    const tree = array2tree([1, 2, 3]);

    expect(diameterOfBinaryTree(tree)).toBe(2);
  });

  test('diameter of binary tree - test case 2', async () => {
    const tree = array2tree([1, 2, 3, 4, 5]);

    expect(diameterOfBinaryTree(tree)).toBe(3);
  });
});

File: test/leetcode/tree/diameter_of_binary_tree_test.py
from leetcode.tree.common.tree_node import array2tree
from leetcode.tree.diameter_of_binary_tree import Solution


soln = Solution()


def test_case_1():
    tree = array2tree([1, 2, 3])

    assert soln.diameterOfBinaryTree(tree) == 2


def test_case_2():
    tree = array2tree([1, 2, 3, 4, 5])

    assert soln.diameterOfBinaryTree(tree) == 3
